From bf6257da37c149e6485e223f9149baea6466407d Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 18 Sep 2025 15:30:59 +0200
Subject: [PATCH] drm/stm: lvds: check pixel clock rate

Null value is forbidden for the pixel clock rate (the PLL clock rate
fails to lock).

Change-Id: I36349c372d28cc101eb09828fbe5242250fd4404
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/491159
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/stm/lvds.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/stm/lvds.c b/drivers/gpu/drm/stm/lvds.c
index 5d95c79ef80b..c02b0f40dede 100644
--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -768,11 +768,12 @@ static int lvds_pixel_clk_register(struct stm_lvds *lvds)
 		 * We want pixclk, not bitclk
 		 */
 		lvds->pixel_clock_rate = (unsigned long)pll_out_khz * 1000 * multiplier / 7;
-	} else {
-		/* set the rate by default at 148500000 */
-		lvds->pixel_clock_rate = 148500000;
 	}
 
+	/* Check the pixel clock rate (null value forbidden) */
+	if (!lvds->pixel_clock_rate)
+		lvds->pixel_clock_rate = 148500000;
+
 	ret = clk_hw_register(lvds->dev, &lvds->lvds_ck_px);
 	if (ret)
 		return ret;
-- 
2.53.0

